src/openforms/formio/formatters/default.py

```diff
@@ -0,0 +1,122 @@
+from typing import Dict
+
+from openforms.plugins.plugin import AbstractBasePlugin
+
+from .registry import register
+
+
+@register("default")
+class DefaultFormatter(AbstractBasePlugin):
```

Comment:
I'd define a base class specific to formatters and inherit from that rather than directly from `AbstractBasePlugin`:

```py
class FormioFormatter(AbstractBasePlugin):
    def __call__(self, component: dict, value: Any) -> str:
        return self.format(component, value)

    def format(self, component: dict, value: Any) -> str:
        raise NotImplementedError("%r must implement the 'format' method" % type(self))
```

The `__call__` approach is neat because you can then clean up calling code something like this:

```py
formatter = register[info["type"]] if info["type"] in register else register["default"]
formatted = formatter(info, info["value"])
```

The whole `info`/`info["value"]` situation in `get_printable_data` is messed up too, since it only pulls a subset of the keys. In my opinion it should just pass the entire formio component definition like how you'd get it from `iter_components` so that the full context is passed to each formatter.